create-api-diff.sh

```diff
@@ -1,121 +1,110 @@
-   curl -m 1200 -sL 'download.eclipse.org/modeling/tmf/xtext/downloads/drops/2.19.0/R201909021322/tmf-xtext-Update-2.19.0.zip' --output 'tmf-xtext-Update-2.19.0.zip' && unzip -q tmf-xtext-Update-2.19.0.zip -d tmf-xtext-Update-2.19.0
-   rm tmf-xtext-Update-2.19.0.zip
-fi
+download () {
+   XTEXT_VERSION=$1
+   DOWNLOAD_URL=$2
+   ZIP_FILE=tmf-xtext-Update-$XTEXT_VERSION.zip
 
-if [ ! -d tmf-xtext-Update-2.20.0 ];
-then
-   echo "Downloading Xtext 2.20.0"
-   curl -m 1200 -sL 'download.eclipse.org/modeling/tmf/xtext/downloads/drops/2.20.0/R201912021256/tmf-xtext-Update-2.20.0.zip' --output 'tmf-xtext-Update-2.20.0.zip' && unzip -q tmf-xtext-Update-2.20.0.zip -d tmf-xtext-Update-2.20.0
-   rm tmf-xtext-Update-2.20.0.zip
-fi
+   if [ ! -d tmf-xtext-Update-$XTEXT_VERSION ]; then
+      echo "Downloading Xtext $XTEXT_VERSION from $DOWNLOAD_URL"
+      # check existence of file
+      if [ $(curl -s -o /dev/null -w '%{http_code}' $DOWNLOAD_URL) == 404 ]; then
+         echo "Not found: $DOWNLOAD_URL"
+         exit 1
+      fi
+      curl -m 1200 -sL $DOWNLOAD_URL --output $ZIP_FILE
+      unzip -q $ZIP_FILE -d tmf-xtext-Update-$XTEXT_VERSION
+      rm $ZIP_FILE
+   fi
+}
 
-if [ ! -d tmf-xtext-Update-2.21.0 ];
-then
-   echo "Downloading Xtext 2.21.0"
-   curl -m 1200 -sL 'download.eclipse.org/modeling/tmf/xtext/downloads/drops/2.21.0/R202003021509/tmf-xtext-Update-2.21.0.zip' --output 'tmf-xtext-Update-2.21.0.zip' && unzip -q tmf-xtext-Update-2.21.0.zip -d tmf-xtext-Update-2.21.0
-   rm tmf-xtext-Update-2.21.0.zip
+# Download latest Xtext build
+if [ -d tmf-xtext-Update-$DEV_VERSION ]; then
+  rm -r tmf-xtext-Update-$DEV_VERSION
 fi
+DOWNLOAD_URL=https://ci.eclipse.org/xtext/job/xtext-umbrella/job/master/lastStableBuild/artifact/build/org.eclipse.xtext.sdk.p2-repository-$DEV_VERSION-SNAPSHOT.zip
+download $DEV_VERSION $DOWNLOAD_URL
 
-if [ ! -d tmf-xtext-Update-2.22.0 ];
-then
-   echo "Downloading Xtext 2.22.0"
-   curl -m 1200 -sL 'download.eclipse.org/modeling/tmf/xtext/downloads/drops/2.22.0/R202006021533/tmf-xtext-Update-2.22.0.zip' --output 'tmf-xtext-Update-2.22.0.zip' && unzip -q tmf-xtext-Update-2.22.0.zip -d tmf-xtext-Update-2.22.0
-   rm tmf-xtext-Update-2.22.0.zip
-fi
+# download NEW_VERSION and OLD_VERSION if not present
+for ((idx=0; idx<${#BUILD_IDS[@]}; ++idx)); do
+   VERSION=${VERSIONS[idx]}
+   BUILD_ID=${BUILD_IDS[idx]}
 
-if [ ! -d tmf-xtext-Update-2.23.0 ];
-then
-   echo "Downloading Xtext 2.23.0"
-   curl -m 1200 -sL 'download.eclipse.org/modeling/tmf/xtext/downloads/drops/2.23.0/R202008310926/tmf-xtext-Update-2.23.0.zip' --output 'tmf-xtext-Update-2.23.0.zip' && unzip -q tmf-xtext-Update-2.23.0.zip -d tmf-xtext-Update-2.23.0
-   rm tmf-xtext-Update-2.23.0.zip
-fi
+   # only download relevant versions
+   if [ $VERSION == $NEW_VERSION ] || [ $VERSION == $OLD_VERSION ]; then
+      download $VERSION "https://download.eclipse.org/modeling/tmf/xtext/downloads/drops/$VERSION/$BUILD_ID/tmf-xtext-Update-$VERSION.zip"
```

Comment:
fix for the wrong version downloaded problem. what do you think @kthoms

Reply:
Looks fine to me